googlemock/test/gmock-actions_test.cc

```diff
@@ -646,6 +647,71 @@ TEST(ReturnRefTest, IsCovariant) {
   EXPECT_EQ(&derived, &a.Perform(std::make_tuple()));
 }
 
+template <typename T, typename = decltype(ReturnRef(std::declval<T&&>()))>
+bool CanCallReturnRef(T&&) { return true; }
+bool CanCallReturnRef(Unused) { return false; }
+
+// Defined here, because gmock has to work with C++11 (std::void_t is from C++17)
+template<typename... Ts> struct precpp17_make_void { typedef void type;};
+template<typename... Ts> using precpp17_void_t = typename precpp17_make_void<Ts...>::type;
+
+template <typename T, typename = void>
+struct HasReturnRefAction : std::false_type {};
+template <typename T>
+struct HasReturnRefAction<T, precpp17_void_t<decltype(ReturnRef(std::declval<T>()))>>
+    : std::true_type {};
+
+// Just an example of non-POD type
+class MyNonPodType {
```

Comment:
Please move these types into the body of the test that uses them to avoid polluting the namespace scope.

Reply:
Ok. To avoid duplicated definitions (repeating the same types in two tests) I just use `int` for scalar types and `std::string` for non-scalar types. Is that OK?